device/src/application/services/impl/DeviceServiceImpl.ts

```diff
@@ -105,25 +116,36 @@ export class DeviceServiceImpl implements DeviceService {
     return await this.repository.getActiveDevices()
   }
 
-  async createDevice(
-    description: string,
-    endpoint: DeviceEndpoint,
-    locationId: string,
-    enabled: boolean,
-    capabilities: DeviceCapability[]
-  ): Promise<DeviceId> {
-    const device: Device = DeviceFactory.deviceFrom(
-      DeviceFactory.newId(),
-      description,
-      endpoint.ipAddress,
-      endpoint.port,
-      locationId,
-      capabilities,
-      enabled
-    )
-    await this.repository.saveDevice(device)
-    this.events.publishDeviceAdded(DeviceEventFactory.createAddition(new Date(), device.deviceId.value))
-    return device.deviceId
+  async createDevice(description: string, endpoint: DeviceEndpoint): Promise<DeviceId> {
+    return await this.servient.start().then(async (WoT: any): Promise<DeviceId> => {
+      const td = await WoT.requestThingDescription(`http://${endpoint.ipAddress}:${endpoint.port}/device`)
+      const thing = await WoT.consume(td)
+      return await thing.readProperty('status').then(async (data: any): Promise<DeviceId> => {
+        const deviceStatus = await data.value()
```

Comment:
@paga16-hash Yes, a "Thing manager" as you mention before would be nice here and there. (A nice to have, let's say)

Reply:
As you prefer. You can take the opportunity to do some TDD if you want. Anyways, you can plan it for a next PR without any problem in my opinion. @Mala1180 what do you think?

Reply:
I think its better to do it in a next PR so i can proceed with general tests and then refactor this part of code. Same for device tests, we have to mock a thing or remove the post test..

Reply:
Next PR is ok